### PR TITLE
Add profiler markers to diags system and visual profiler

### PR DESCRIPTION
**Pull request by davidkline-ms**
This is the part of #7565 that adds unity profiler instrumentation to the diagnostics system and the visual profiler.

### CODE REVIEW

File: Assets/MRTK/Services/DiagnosticsSystem/MixedRealityToolkitVisualProfiler.cs [hunk 1]
```diff
@@ -805,23 +846,38 @@ private static ulong AppMemoryUsageLimit
...
+                return (int)(oldUsageMBs * decimalPower) != (int)(newUsageMBs * decimalPower);
+            }
         }
 
+        private static readonly ProfilerMarker ConvertMegabytesToBytesPerfMarker = new ProfilerMarker("[MRTK] MixedRealityToolkitVisualProfiler.ConvertMegabytesToBytes");
+
         private static ulong ConvertMegabytesToBytes(int megabytes)
         {
-            return ((ulong)megabytes * 1024UL) * 1024UL;
+            using (ConvertMegabytesToBytesPerfMarker.Auto())
```
Comment (Cameron-Micka) [inline Assets/MRTK/Services/DiagnosticsSystem/MixedRealityToolkitVisualProfiler.cs:867]: nit: Do we need markers this granular?

File: Assets/MRTK/Services/DiagnosticsSystem/MixedRealityToolkitVisualProfiler.cs [hunk 2]
```diff
@@ -805,23 +846,38 @@ private static ulong AppMemoryUsageLimit
...
+                return ((ulong)megabytes * 1024UL) * 1024UL;
+            }
         }
 
+        private static readonly ProfilerMarker ConvertBytesToMegabytesPerfMarker = new ProfilerMarker("[MRTK] MixedRealityToolkitVisualProfiler.ConvertBytesToMegabytes");
+
         private static float ConvertBytesToMegabytes(ulong bytes)
         {
-            return (bytes / 1024.0f) / 1024.0f;
+            using (ConvertBytesToMegabytesPerfMarker.Auto())
```
Comment (Cameron-Micka) [inline Assets/MRTK/Services/DiagnosticsSystem/MixedRealityToolkitVisualProfiler.cs:877]: nit: Do we need markers this granular?

File: Assets/MRTK/Services/DiagnosticsSystem/MixedRealityToolkitVisualProfiler.cs (hunk 1 above)
Comment (wiwei) [reply Assets/MRTK/Services/DiagnosticsSystem/MixedRealityToolkitVisualProfiler.cs:867]: Yeah some of these feel over instrumented (i.e. especially instrumenting itoa())

Comment (davidkline-ms) [reply Assets/MRTK/Services/DiagnosticsSystem/MixedRealityToolkitVisualProfiler.cs:867]: i'll remove

Comment (davidkline-ms) [reply Assets/MRTK/Services/DiagnosticsSystem/MixedRealityToolkitVisualProfiler.cs:867]: done

File: Assets/MRTK/Services/DiagnosticsSystem/MixedRealityToolkitVisualProfiler.cs (hunk 2 above)
Comment (davidkline-ms) [reply Assets/MRTK/Services/DiagnosticsSystem/MixedRealityToolkitVisualProfiler.cs:877]: done

**Pull request merged**